### PR TITLE
[@builder.io/gatsby] update to gatsby 4 and use graphql-source to fetch data for batching and  retries

### PR DESCRIPTION
**Pull request by teleaziz**

### CODE REVIEW

**Comment by teleaziz** [comment]
Ideally this plugin is built using the gatsy-source-graphql as in this commit https://github.com/BuilderIO/builder/pull/762/commits/cd5da833d972515a18ab0c16f388db1171632b5d , unfortunately this is blocked by this issue https://github.com/gatsbyjs/gatsby/issues/34042 so opt it to make a temporary fork until the pr with the fix is merged

**Review by mrkoreye — APPROVED** [review]

**Pull request merged**
